src/components/Authentication/Authentication.js

```diff
@@ -1,28 +1,56 @@
 import React from 'react';
 import PopupWithForm from '../PopupWithForm/PopupWithForm.js';
+import { useFormWithValidation } from '../FormValidation/FormValidation.js';
 
 function Authentication(props) {
+    const { values, handleChange, isValid, resetForm } = useFormWithValidation({});
+
+    const submitForm = (e) => {
+        e.preventDefault();
+        const { email, password } = values;
+        props.formSubmit(email, password);
+        resetForm();
+    }
+
+
     return (
         <PopupWithForm
-            auth={props.auth}
+            altFormLink={props.altFormLink}
+            altFormTitle={props.altFormTitle}
             buttonText='Sign in'
+            formSubmit={submitForm}
+            isOpen={props.isOpen}
+            isValid={isValid}
             title='Sign in'
             type='auth'
-            altForm='Sign up'
-            isOpen={props.isOpen}
-            onClose={props.onClose}
             openAltForm={props.openAltForm}
+            onClose={props.onClose}
         >
             <h6 className='popup__input-title popup__input-title_type_email'>Email</h6>
-            <input minLength='1' maxLength='30' type='text'
-                className='popup__input popup__input_type_email' placeholder='Email' />
-            <span className='popup__error'>Ivalid email address</span>
+            <input
+                minLength='1'
+                maxLength='30'
+                type='email'
+                className='popup__input popup__input_type_email'
+                placeholder='Email'
+                onChange={handleChange}
+                name='email'
+            />
+            <span className={`popup__error ${isValid === false && values.email !== undefined && 'popup__error_visible'}`}>Invalid email address</span>
 
             <h6 className='popup__input-title popup__input-title_type_password'>Password</h6>
-            <input minLength='1' maxLength='30' type='text'
+            <input
+                minLength='1'
+                maxLength='30'
+                type='text'
                 className='popup__input popup__input_type_password'
-                placeholder='Enter password' />
-        </PopupWithForm>
+                placeholder='Enter password'
+                name='password'
+                required
+                onChange={handleChange}
```

Comment:
If you have `onChange` in an input then you need to add attribute `value` to this input.
This is how React controls inputs.
To read more about `controlled components`: https://reactjs.org/docs/forms.html#controlled-components